alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/SolrMetricsBeanPostProcessor.java

```diff
@@ -0,0 +1,79 @@
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.config.ConstructorArgumentValues;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+
+public class SolrMetricsBeanPostProcessor implements BeanDefinitionRegistryPostProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrMetricsBeanPostProcessor.class);
+
+    public static final String SOLR_SHARDING_REGISTRY_BEAN_ID = "search.SolrShardRegistry";
+    public static final String SOLR_SHARDING_METRICS_BEAN_ID = "eu.xenit.alfred.telemetry.binder.solr.sharding.SolrShardingMetrics";
+    public static final String SOLR_SHARDING_METRICS_CRON_PROPERTY = "eu.xenit.alfred.telemetry.binder.solr.sharding.cronexpression";
+    public static final String SOLR_SHARDING_METRICS_ENABLED_PROPERTY = "eu.xenit.alfred.telemetry.binder.solr.sharding.enabled";
+    public static final String QUARTZ_CHECK_CLASS = "org.quartz.ScheduleBuilder";
+    private final Scheduler scheduler;
+    private final Properties globalProperties;
+
+    private MeterRegistry meterRegistry;
+
+    public SolrMetricsBeanPostProcessor(Properties globalProperties, MeterRegistry meterRegistry, Scheduler scheduler) {
+        this.globalProperties = globalProperties;
+        this.meterRegistry = meterRegistry;
+        this.scheduler = scheduler;
+    }
+
+    @Override
+    public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry beanDefinitionRegistry) throws BeansException {
+        if (!Boolean.parseBoolean(globalProperties.getProperty(SOLR_SHARDING_METRICS_ENABLED_PROPERTY))) {
+            LOGGER.info("Solr sharding metrics are not enabled, skipping.");
+            return;
+        }
+
+        BeanDefinition solrShardingRegistryBean;
+        try {
+            solrShardingRegistryBean = beanDefinitionRegistry.getBeanDefinition(SOLR_SHARDING_REGISTRY_BEAN_ID);
+        } catch (NoSuchBeanDefinitionException e) {
+            LOGGER.info("Bean with id {} not found in be, sharding metrics will not be made available.");
+            return;
+        }
+
+        try {
+            Class.forName(QUARTZ_CHECK_CLASS);
+        } catch (ClassNotFoundException e) {
+            LOGGER.info("{} not found on classpath. Sharding metrics will not be made available.");
+            return;
+        }
+
+        GenericBeanDefinition solrShardingMetricsBean = new GenericBeanDefinition();
+        ConstructorArgumentValues constructorArgumentValues = new ConstructorArgumentValues();
+        // referencing the bean by name does not seem to work in this context
```

Comment:
:exploding_head: strange

Reply:
Yes, but it does not look ugly this way. I think it has something to do with subsystem contexts.